CarpetCleaning/StoredProcedures/Dbo.validateexceldata.sql

```diff
@@ -0,0 +1,126 @@
+CREATE PROCEDURE ValidateExcelData
+	
+AS
+BEGIN
+	
+	SET NOCOUNT ON;
+
+	DECLARE
+		@MissingInformationError NVARCHAR(255) = 'Missing information in one or more fields',
+		@DuplicateInfoError NVARCHAR(255) = 'Duplicate of invoice',
+		@NoServicesError NVARCHAR(255) = 'No services performed',
+		@TotalNotMatchError NVARCHAR(255) = 'Amount of services combined does not equal total',
+		@NotValidTechnician NVARCHAR(255) = 'Cannot find Technician in table',
+		@NotValidCity NVARCHAR(255) = 'City not in Table',
+		@NotValidState NVARCHAR(255) = 'State not in Table',
+		@WrongZipcode NVARCHAR(255) = 'Wrong number of digits in zip code',
+		@PassedValidation NVARCHAR(255) = 'Passed Validation',
+		@FailedValidation NVARCHAR(255) = 'Failed Validation'
+
+	--Everytime Validation is run, clean all previous errors off
+	TRUNCATE TABLE DBO.ExcelDataError
+
+		--Clean up empty spaces
+	UPDATE DBO.ExcelData
+	SET TechnicianName = LTRIM(RTRIM(Technicianname)),
+	Addressnumber = LTRIM(RTRIM(Addressnumber)),
+	Street = LTRIM(RTRIM(Street)),
+	Apt = LTRIM(RTRIM(Apt)),
+	City = LTRIM(RTRIM(City)),
+	[State] = LTRIM(RTRIM(Technicianname))
+
+	--Missing Information
+	INSERT INTO DBO.ExcelDataErrors (ExcelDataId, ErrorMessage)
+		SELECT RowId, @MissinginformationError 
+		FROM DBO.ExcelData
+			WHERE (TechnicianName = '' OR TechnicianName IS NULL)
```

Comment:
This is a good example.  Just bring the where back.  Indentation implies that something belongs inside whatever is above it.  There WHERE doesn't belong inside the select statement, it's just part of the select statement.  So let's bring the indentation on this WHERE back to the same place as the select and the from.